src/docfx/build/xref/XrefResolver.cs

```diff
@@ -162,15 +144,26 @@ private string RemoveSharingHost(string url, string hostName)
             return url;
         }
 
-        private (Error?, IXrefSpec?) Resolve(SourceInfo<string> uid, Document referencingFile)
+        private (Error?, IXrefSpec?, string? href) Resolve(SourceInfo<string> uid, Document referencingFile, Document inclusionRoot, string query = "")
         {
             var unescapedUid = Uri.UnescapeDataString(uid);
             var xrefSpec = ResolveInternalXrefSpec(unescapedUid, referencingFile) ?? ResolveExternalXrefSpec(unescapedUid);
             if (xrefSpec is null)
             {
-                return (Errors.Xref.XrefNotFound(uid), null);
+                return (Errors.Xref.XrefNotFound(uid), null, null);
+            }
+
+            var href = UrlUtility.MergeUrl(RemoveSharingHost(xrefSpec.Href, _config.HostName), query);
+
+            // NOTE: this should also be relative to root file
+            var sourceFile = inclusionRoot ?? referencingFile;
+            _fileLinkMapBuilder.AddFileLink(sourceFile.FilePath, sourceFile.SiteUrl, href);
```

Comment:
By moving this logic down to `Resolve`, the file link will be added for `ResolveXrefSpec` as well, was there a missing dependency for `xref` node in SDP before?

Reply:
I had another look, maybe this add file link and calculating relative herf should belong to method `ResolveInternalXrefSpec`, we should not have them for external UID, right?

Reply:
It's always added before.
https://github.com/dotnet/docfx/blob/v3/src/docfx/build/xref/XrefResolver.cs#L86

Reply:
@live1206  Move relative/removeHost logic inside resolve internal/external